host/tests/deepsleep_L3_Tests/test1_TriggerDeepsleep.py

```diff
@@ -0,0 +1,382 @@
+            sock.send("Hello from Raspberry Pi!\n")  # Send some text
+            time.sleep(1)
+            sock.send("\r")  # Send an Enter keypress
+
+            # Close the connection
+            sock.close()
+
+    def wake_on_presence_detection(self, manual=False):
+        """
+        Triggers a wake-up event based on presence detection.
+
+        Args:
+            None.
+        """
+        if manual == True:
+            return self.testUserResponse.getUserYN(f"Please trigger wake up via presence detection. Did the device wakeup as expected (Y/N):")
+
+        else:
+            #TODO: Add automation wakeup and verification methods
+            return False
+
+    def wake_on_power_key(self, manual=False):
+        """
+        Triggers a wake-up event based on the power key.
+
+        Args:
+            None.
+        """
+        if manual == True:
+            return self.testUserResponse.getUserYN(f"Please trigger wake up via power key. Did the device wakeup as expected (Y/N):")
+
+        else:
+            #TODO: Add automation wakeup and verification methods
+            return False
+
+    def wake_on_IR(self, manual=False):
+        """
+        Triggers a wake-up event based on IR signal.
+
+        Args:
+            None.
+        """
+        if manual == True:
+            return self.testUserResponse.getUserYN(f"Please trigger wake up via IR. Did the device wakeup as expected (Y/N):")
+
+        else:
+            #TODO: Add automation wakeup and verification methods
+            return False
+
+    def testFunction(self):
+        """This function will test the get cpu tempature functionality
+
+        Returns:
+            bool
+        """
+        # Create the deepsleep manager class
+        self.testDeepsleep = deepsleepClass(self.deviceProfile, self.hal_session_deepsleep)
+
+
+        # Create the Power Manager Class
+        self.testPower = powerManagerClass(self.deviceProfile, self.hal_session_power)
+
+        # Initialize the deepsleep manager and power manager
+        self.testPower.initialise()
+        self.testDeepsleep.initialise()
+
+        self.wakeupsources = self.testPower.getWakeupSources()
+
+
+        result = True
+
+        finalResult = True
+
+        for source in self.wakeupsources:
+            self.log.stepStart(f"Triggering wake up test for {self.testPower.wakeup_sources_map[source]}")
+
+            self.log.step(f"Setting wakeup source to {self.testPower.wakeup_sources_map[source]}")
+            self.testPower.setWakeupSource(source, 1)
+
+            self.log.step(f"Triggering deepsleep")
+            if source == 0x06:
+                self.testDeepsleep.triggerDeepsleep(60, 1)
+            else:
+                self.testDeepsleep.triggerDeepsleep(0, 1)
+
+            if source == 0x00:
+               # Wake on Voice
+               result = self.wake_on_voice("rtmp://your-backend-server/live/stream", "audio.wav", True)
+               if result:
```

Comment:
this code is repeating for all the verification, add once after all the elif 

```
 if result:
                   self.log.stepResult(result, f"Voice wakeup success")
               else:
                   self.log.stepResult(result, f"Voice wakeup failed")
               pass

```

replace with

```
self.log.stepStart(f"Triggering wake up test for {self.testPower.wakeup_sources_map[source]}")
if
elif
elif
elif
:
:
 self.log.stepResult(result, f"Wake up test for {self.testPower.wakeup_sources_map[source]}")
```

Reply:
Updated